datajunction-server/datajunction_server/api/metrics.py

```diff
@@ -96,13 +98,20 @@ async def list_metric_metadata() -> MetricMetadataOptions:
 
 @router.get("/metrics/{name}/", response_model=Metric)
 async def get_a_metric(
-    name: str, *, session: AsyncSession = Depends(get_session)
+    name: str,
+    *,
+    session: AsyncSession = Depends(get_session),
+    application_cache: Optional[CacheInterface] = Depends(get_cache),
```

Comment:
Why `application_cache` and not simply `cache`?

Reply:
I initially had that but it conflicted with the `fastapi_cache.decorator.cache` that we use as `@cache` on the server endpoints which is just caching responses based on the request signature. It made me realize we may get these caches confused in the future so went for a more explicit `application_cache` variable name so we know this is the dependency injected cache to do caching/cache-invalidation within the application logic.

Reply:
The risk of confusing them is non-trivial and could lead to some unexpected behavior.